gnome/extension.js

```diff
@@ -117,11 +117,26 @@ const ContestCountdown = new Lang.Class({
   // TODO: Issue #4
   // Make the string to display in top bar using contests objects
   // @return: string
-  _getTimerText: function () {},
+  _getTimerText: function () {
+    var timerText;
+
+    let timeDiff = this.contests.secondsTillNextContest();
+
+    if (timeDiff >= 0) {
```

Comment:
+infinity will be handled by this if.

Note: If you find the edge cases given by secondsTillNextContest unwieldy. You can edit them to -1 -2 -3.. or anything. Just make sure to document them properly.